### PR TITLE
Removing vehicle manager from vehicles and renaming energy conservation params

### PR DESCRIPTION
**Pull request by haitamlaarabi**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3443)

### CODE REVIEW

**Comment by haitamlaarabi** [comment]
Test now!

File: src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala [hunk 1]
```diff
@@ -725,7 +722,7 @@ trait ChoosesMode {
               seq :+ (vehicleOnTrip -> ParkingInquiry.init(
                 SpaceTime(geo.wgs2Utm(leg.beamLeg.travelPath.endPoint.loc), leg.beamLeg.endTime),
                 nextAct.getType,
-                VehicleManager.getReservedFor(veh.vehicleManagerId.get).get,
+                VehicleManager.getReservedFor(this.vehicleManagerId).get,
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala:725]: What if we use a shared vehicle that has its own vehicle manager? In this case, we receive a wrong parking stall.

File: src/main/scala/beam/agentsim/agents/parking/ChoosesParking.scala [hunk 2]
```diff
@@ -113,7 +113,7 @@ trait ChoosesParking extends {
     val parkingInquiry = ParkingInquiry.init(
       SpaceTime(destinationUtm, lastLeg.beamLeg.endTime),
       nextActivityType,
-      VehicleManager.getReservedFor(currentBeamVehicle.vehicleManagerId.get).get,
+      VehicleManager.getReservedFor(this.vehicleManagerId).get,
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/parking/ChoosesParking.scala:116]: Again: what if we use a shared vehicle here?

**Review by haitamlaarabi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3443)* status: 0 of 32 files reviewed, 2 unresolved discussions (waiting on @dimaopen and @haitamlaarabi)

___
*[src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala, line 725 at r1](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3443#-MtREQFeeAz4HM7ayTnG-r1-725:-MtqPM6dAcoF6ZCsjijQ:b5uhlg6) ([raw file](https://github.com/lbnl-ucb-sti/beam/blob/42969f980bdbf583f34e042d9099c57f4a7b083b/src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala#L725)):*
<details><summary><i>Previously, dimaopen (Dmitrii Openkov) wrote…</i></summary><blockquote>

What if we use a shared vehicle that has its own vehicle manager? In this case, we receive a wrong parking stall.
</blockquote></details>

This is a good point. Let's not merge this until I figure out whether there is a workaround or we just have to keep it as it is!

**Comment by haitamlaarabi** [comment]
Do not merge #WIP

**Comment by JustinPihony** [comment]
Bumping to a draft since it is a WIP

**Pull request closed without merging**